internals/secrethub/variable_reader_test.go

```diff
@@ -0,0 +1,144 @@
+package secrethub
+
+import (
+	"testing"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+
+	"github.com/secrethub/secrethub-cli/internals/secrethub/tpl"
+	"github.com/secrethub/secrethub-go/internals/assert"
+)
+
+func TestVariableReader(t *testing.T) {
```

Comment:
Maybe add a testcase where is asserted that variables supplied for the specific command have preference over the variables supplied via the `SECRETHUB_VAR_<KEY>` envvar shared between `run` and `inject`.

This is a case where a key is supplied in both `osEnv` and `commandTemplateVars`. The value from `commandTemplateVars` should be used.